### ISSUE TITLE
typeinfo.h No such file or directory

### ISSUE THREAD

**Issue by Martymoose98** [ValveSoftware/source-sdk-2013#529]
While trying to make a server plugin I encountered some build errors when building the Serverplugin_empty project.
I looked online and some results said to use `#include <typeinfo>`. If I do that I get a bunch of various errors on with the c runtime library however, I did update the Platform Toolset to Visual Studio 2022 (v143) from Visual Studio 2013 - Windows XP (v120_xp) (not installed) because I couldn't find it in the vs installer, maybe I am blind. Sorry if this is a bad report or caused because of the toolset change. It appears to me that `_CRTNOALIAS` has no definition else it would be purple and using "Peek Definition" results in nothing.
![image](https://user-images.githubusercontent.com/20148294/147729413-e6f8cb58-01a9-42b7-9806-890f69cde1da.png)

```
Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	E1696	cannot open source file "typeinfo.h"	Serverplugin_empty	\source-sdk-2013\mp\src\public\tier0\memalloc.h	387	


Severity	Code	Description	Project	File	Line	Suppression State
Error	C2086	'int _CRTNOALIAS': redefinition	Serverplugin_empty	\source-sdk-2013\mp\src\public\tier0\memoverride.cpp	132	
```

**Comment by TheASVigilante** [comment]
Try includeing it in upper casing like this:
#include "TypeInfo.h"
if it doesn't work just try replacing <> with ""

**Comment by TheASVigilante** [comment]
remove the .h at the end of it.

**Comment by geotavros** [comment]
You can save yourself time and just download VS2013:

https://visualstudio.microsoft.com/vs/older-downloads/

**Comment by RealFX-Code** [comment]
> You can save yourself time and just download VS2013:
> 
> https://visualstudio.microsoft.com/vs/older-downloads/

But is there a fix for VS community 2017/2022?

**Comment by brannanz** [comment]
VS2013 worked like a charm, but the servers are down so you cannot activate it so VS softlocks itself, and source SDK craps out on compile with new visual studio.

**Comment by Martymoose98** [comment]
Yeah, it is strange I guess you might be able to use the I think 2014 CS:GO leaked source maybe that compiles better? Perhaps using the allied modders SDK as an alternative? Would be sick if someone at valve could chime in though.  If I remember right, I caved and downloaded vs2013 which was a real pain because Microsoft barely supports vs2013 anymore (it is a separate download not in the installer). The installation process also switched all .h, .c, .cpp, etc... files to open with the vs2013. Which was a delight to fix. When I got it working I believe there still was linker issues with Msg in tier0. However, my memory might be duping me as it could have been the wrong branch of the  allied modders SDK. Was like a few months ago, what I do remember it was tedious and vexing to work with.
